pkg/virt-controller/watch/util/pendingqueue.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Contributors
+ *
+ */
+
+package util
+
+import (
+	"sync"
+
+	"k8s.io/client-go/util/workqueue"
+)
+
+type PendingQueue struct {
+	queue     []string
+	queueLock *sync.Mutex
+}
+
+func NewPendingQueue() *PendingQueue {
+	return &PendingQueue{
+		queue:     []string{},
+		queueLock: &sync.Mutex{},
+	}
+}
+
+func (pq *PendingQueue) GetQueue() []string {
+	pq.queueLock.Lock()
+	defer pq.queueLock.Unlock()
+	return pq.queue
+}
```

Comment:
You're handing out the original slice here rather than a copy, so the lock is redundant, or you should switch to returning a copy.

In addition, maybe it's just me, but I find the function name a bit confusing, since `queue` usually means a queue object (like `RateLimitingInterface`) although here it's a string slice, but that's an implementation detail.

WDYT about
```suggestion
func (pq *PendingQueue) GetKeys() []string {
	pq.queueLock.Lock()
	defer pq.queueLock.Unlock()
	return slices.Clone(pq.queue)
}
```

Reply:
I almost made the same comment to Federico when he suggested this, but decided to test it in the Go playground, and the returned slice does appear to be a copy, unaffected by changes to the original slice.
Maybe slices are passed as "references" when calling a function, but copied on return?
Either way, as discussed, this function is only used in unit tests, where the lock is not necessary.

Reply:
https://play.golang.com/p/wl2VYExfZSg
Are you sure? :)

Code:
```go
type obj struct {
	s []string
}

func getObj() obj {
	return obj{
		s: []string{"one", "two"},
	}
}

func (o *obj) getSlice() []string {
	return o.s
}

func main() {
	// Initiate an object with a slice
	o := getObj()
	
	// Return the slice with copying it
	s := o.getSlice()
	
	// Change a slice element
	s[0] = "ten"
	
	// Print the object's internal slice
	fmt.Println(o.getSlice())
}
```

Output:
```
[ten two]
```